### PR TITLE
Added routing skeleton to server

### PR DESCRIPTION
**Pull request by jlforcherio1**

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -1,20 +1,108 @@
...
+const app = express();
 
+const port = 3000;
 
-const port = 3000
+app.use(express.static(path.join(__dirname, "../public")));
 
-app.use(express.static(path.join(__dirname, '../public')));
+app.get("/", (req, res) => {
+  res.send("Hello World!");
```
Comment (xil324) [inline server/index.js:10]: Could you remove those lines since we don't have an endpoint '/'?

File: server/index.js [hunk 2]
```diff
@@ -1,20 +1,108 @@
...
+  // parse response into example object
+  // send example object back to the front-end
+  res.send("getRecipesFromIngredients");
+});
 
-app.get('/', (req, res) => {
-  res.send('Hello World!')
-})
+app.get("/getUserInfo", (req, res) => {
+  // request should include user id
```
Comment (xil324) [inline server/index.js:22]: For user id, are we using uid?

File: server/index.js [hunk 3]
```diff
@@ -1,20 +1,108 @@
...
 
+app.get("/getFeaturedRecipes", (req, res) => {
+  // queries database for most upvoted recipes
+  // send ids to spoonacular
+  // send back most upvoted recipes
+  res.send("getFeaturedRecipes");
+});
 
+app.post("/createUser", (req, res) => {
+  // request should include UID from firebase
```
Comment (xil324) [inline server/index.js:36]: Based on our conversation earlier, the request body also includes username and profile picture link.

File: server/index.js [hunk 4]
```diff
@@ -1,20 +1,108 @@
...
+  // query database to create a new user
+  // send "successfully created new user"
+  res.send("createUser");
+});
 
+app.get("/authenticate", (req, res) => {
+  // request should include UID from firebase
```
Comment (xil324) [inline server/index.js:43]: I think we need the ID token to send a request to firebase to verify user on the server side. However, we are not using the Firebase client app communications. I can either send the uid or I can send the iD token and you can verify the ID token using Firebase Admin SDK to get the uid. (https://firebase.google.com/docs/auth/admin/verify-id-tokens) 

"If your Firebase client app communicates with your backend server, you might need to identify the currently signed-in user on your server so you can perform server-side logic on their behalf. You can do this securely by using ID tokens, which are created by Firebase when a user signs into an Firebase app."  (https://firebase.google.com/docs/auth/admin)

File: server/index.js [hunk 5]
```diff
@@ -1,20 +1,108 @@
...
+});
+
+app.put("/updateFavorites", (req, res) => {
+  // request should include user id and recipeId
+  // query database to update note
+  res.send("updateFavorites");
+});
+
+app.put("/updateUsername", (req, res) => {
+  // request should include user id and recipeId
```
Comment (xil324) [inline server/index.js:95]: Are we using uid or user id?

File: server/index.js [hunk 6]
```diff
@@ -1,20 +1,108 @@
...
+  // query database to update note
+  res.send("updateUsername");
+});
+
+app.put("/updateProfilePic", (req, res) => {
+  // request should include user id and recipeId
```
Comment (xil324) [inline server/index.js:101]: same question as above

**Review by xil324 — COMMENTED** [review]
Hey Johnathan,
Thanks for putting all the api endpoints together. I left some comments below.  let me know what your thoughts are.

File: server/index.js (hunk 1 above)
Comment (jlforcherio1) [reply server/index.js:10]: Sure!

File: server/index.js (hunk 4 above)
Comment (jlforcherio1) [reply server/index.js:43]: Could you send the uid? I still want to verify the client, but I'd like to check equality with the uid sent from the front end and the uid sent from firebase on the server side just to be extra safe

**Pull request merged**